tests/TESTING.md

```diff
@@ -0,0 +1,105 @@
+# Testing Standards in K3s
+
+Go testing in K3s comes in 3 forms: Unit, Integration, and End-to-End (E2E). This
+document will explain *when* each test should be written and *how* each test should be
+generated, formatted, and run.
+
+Note: all shell commands given are relateive to the root k3s repo directory.
+___
+
+## Unit Tests
+
+Unit tests should be written when a complex component or function of a package needs testing.
```

Comment:
I'm not sure we want to say "complex" here as there are going to be trivial and maybe more obvious changes made that are in the critical path that also need testing.